### DISCUSSION TITLE
Hac Mini for Sonoma and Sequoia

### DISCUSSION THREAD

**Discussion by goodbest** [osy86/HaC-Mini#1091] [General]
Just updated some kext and added support for Sonoma 14.0-14.6 official release, as well as Sequoia 15.0 beta.

You can download the pkg [here](https://github.com/goodbest/HaC-Mini/releases/download/20240807/HaCMini.pkg)

# What is tested working
- Metal graphics, audio, etc
- Wifi/BT(see below for further instructions)
- 2nd ethernet (you must select `Intel I210(experiement)` in the installer.

# What is not tested, but may work.
- sd card
- iServices
- TB3 devices

![Screenshot 2024-08-07 at 13 20 54](https://github.com/user-attachments/assets/e3656a6f-9d7a-4e67-9cc8-579d66e251c0)
# Wifi support details
###  Sonoma & Sequoia
1. Select the option in Hac-mini installer
    - Wifi Patch (sonoma+)
2. After Hac-mini installation and OS reboots, run root patch in latest OpenCore Legacy Patcher, and reboots again. 
    - [OpenCore Legacy Patcher v2.0 and above](https://github.com/dortania/OpenCore-Legacy-Patcher/releases)
    - It's easy to patch. You can read the thread [here](https://elitemacx86.com/threads/how-to-fix-broadcom-wifi-on-macos-sonoma-and-later.1415/) (staring from Step 9)

###  Sonoma and Sequoia
1. Select the option in Hac-mini installer
    - Wifi Patch (sonoma+)
    - Dell DW1820A
2. After Hac-mini installation and OS reboots, run root patch in latest OpenCore Legacy Patcher, and reboots again.
    - [OpenCore Legacy Patcher v2.0 and above](https://github.com/dortania/OpenCore-Legacy-Patcher/releases)
    - It's easy to patch. You can read the thread [here](https://elitemacx86.com/threads/how-to-fix-broadcom-wifi-on-macos-sonoma-and-later.1415/) (staring from Step 9)

## Intel Wifi/BT
Airdrop and other continuity features are very limited for Intel Cards.
###  Sonoma
1. Select the option in Hac-mini installer
    - Intel Bluetooth 
    - Intel Wireless
2. DO NOT select:
    - Wifi Patch (sonoma+)
    - Dell DW1820A
###  Sequoia
As of August 2024,  intel wifi kext for Sequoia is not yet released in [this repo](https://github.com/OpenIntelWireless/itlwm)
  - If you can wait, you can wait for the AirportItlwm kext release for Sequoia. It's up to the kext maintainer. Maybe it will be the Fall 2024.
  - If you can't wait, use itlwm+Heliport instead.

**Comment by SashonIT** [comment]
Alright @goodbest . I will Test it in the next days.  For WIFI we have to stay tuned. The Problem is still everywhere. Their is a lot Cooking at the Moment to get this running.  Maybe Later with a KExt or Patch from OCLP Team.  They are Hard Working on it.

**Comment by 1git** [comment]
Looks like the workarounds in @goodbest 's post have gone live... Now I hope someone can figure out how to use them with our opencore environment... I miss my Broadcom wifi :)  The rest of Sonoma seems pretty stable... great work @goodbest !

**Comment by sxunix** [comment]
It is working for the machine type MacMini. However, when I changed the machine type as iMac 2019, it cannot enter the login page. Would you please help to check it? Thanks a lot.

**Comment by SashonIT** [comment]
To get wifi up and running (only for Broadcom Wifi Cards) please read this Workaround here.  It's in German. But you can Translate viA deepl ODER gOOGLE tRANSLATE:

Fazit:

After Sonoma fresh install (complete Installer) You need to Patch via Open Core Legacy Patcher Version minimum 0.6.9.

But two days ago Version 1.0.0 came out. And Yesterday Version 1.0.1  With this Patcher you can apply a root Patch to get Wifi / Bluetooth / Handoff / Continuity / Air Drop up and running.

https://github.com/dortania/OpenCore-Legacy-Patcher

Here is the guide from the german Hackintosh Forum.

https://www.hackintosh-forum.de/forum/thread/58656-anleitung-f%C3%BCr-broadcom-wlan-patch-unter-macos-sonoma/?pageNo=1

**Reply by 1git** [reply]
Thanks, That worked for me! Sonoma; upgraded; added the Broadcom Kexts (don't forget this part...) and the wifi patch kexts; updated Opencore to the latest, ran the post install tool and it all seems to work!

The SD card reader, didn't read a 64 GB SD Card, but I use a USB adapter these days because that got flakey a long time ago... 
I don't have any TB3 devices so couldn't test that either...


Thanks all for your combined effort!

**Reply by SashonIT** [reply]
For TB 3.  I have a Antelope Thunderbolt 3 Audio Interface.

So i can Test it, when i have Sonoma ready on my HacMini.

Can you share your EFI.....  That would be very Kind of you ..... so i can prepare directly a Bootstick with your EFI.

Only WIFI Patching afterwards.

Then i can Test TB3 during the next days.

DISCLAIMER: Antelope AUDIO Announced on their Website that they still working on the driver for Sonoma.

But i will give it a try.

**Comment by SashonIT** [comment]
I would now like to inform you that I have successfully installed Sonoma 14.1.2 on the fine little NUC 8I7HVK in the last three days.
The whole thing was a bit tricky.
But now everything is running to my satisfaction.
Bluetooth, WIFI, Airdrop, Continuity, HandOff and system updates are all working so far.
Tomorrow I'll test my Thunderbolt 3 audio interface and report back.
Anyone interested in my Efi is welcome to contact me.
Or if you want to know how I did the whole thing.
So thanks to everyone here.

**Reply by jeffe** [reply]
Would absolutely be curious to know process

**Reply by SashonIT** [reply]
First I created an installer USB stick with Sonoma 14.1.  installer on it.
Then I used the HacMini 3.3.1 installer from GoodBest as the basis for the whole thing and installed it on the stick.
Then I installed Sonoma.
Now the fun really started.
Fortunately, there is enough expertise and instructions on how to patch the WIFI in the German Hackintosh forum.
Here is the link to what you need to do.
https://www.hackintosh-forum.de/forum/thread/58656-anleitung-f%C3%BCr-broadcom-wlan-patch-unter-macos-sonoma/?pageNo=1
Insert 4 Kexts into your OpenCore configuration in the correct order and revise the boot args and insert the necessary arguments.
Sequence of the chain ---> See picture
1. lilu
2. amfipass.kext
3. etc.
Oh well.
To patch WIFI with OCLP 1.2.1, SIP must be switched off.
There is the (Toggle SIP) driver in OpenCore for this.
SIP must then unfortunately remain permanently off, otherwise no WIFI.
Finally, the restrictevents.kext must be inserted in Opencore.
With the boot argument --> "revpatch=sbvmm"
Reboot ---> NVRAM RESET
AND then you are done.
By the way ---> doing a NVRAM reset after every reboot is helpful.

Translated with DeepL.com (free version)
<img width="1296" alt="image" src="https://github.com/osy/HaC-Mini/assets/126977959/bc466948-b6aa-4c61-b199-0e19d787d197">

**Comment by immie-jay** [comment]
Could you please upload your EFI

**Comment by SashonIT** [comment]
Sonoma Behaviour Collection!

Sleep is not really working.   ---> sometimes it works.  Sometimes it causes a Kernel Panic.

Also when I browse with Safari and change a Tab or clicking somewhere. A RedFrame pops up.

**Reply by 1git** [reply]
Also remember because SIP is off, you need to manually update OS... no easy patches anymore...

**Reply by SashonIT** [reply]
Here is my experience report on the Sonoma 14.2 update
I can tell you briefly about my odyssey yesterday.

I couldn't get the revert from the patcher to start.

Because there were SIP interventions in three places.

1. SIP TOGGLE EFI driver.
2. csr-active-config on 03080000
3. OCLP also does something there.

I actually had to set csr-active-config = 03080000 to = FF0F0000.

And as MacGrummel already wrote, the Luke Skywalker Kext family must be deactivated. Just like Amfipassbeta.kext and the associated Boot-Arg EIntrag "-amfipassbeta" must be temporarily removed.

The good Mr al6042 gave me the tip that it is best to create two configs so that the work is only done once.

One for the normal system with W-Lan.

And one for updating from the system.

With my box, it took a long time to boot up when the Amfipass and the Skywalker family were out of the house.

But in the end it worked.

Then I could finally revert the patcher ---> Then perform the update. ( quite a lot of restarts)

After a successful update, the roll backwards again.

Start PAtcher and install the W-Lan patch again.

Then reactivate the KExte and set csr-active-config = 03080000.

Or simply insert the corresponding config.plist.

What a ride.

**Reply by SashonIT** [reply]
What I forgot to mention.

Everything runs very stable for me, except for the SD card reader.

Maybe someone can help me and work with me on a solution.

If any of you are interested.

Well, I would like to.

Thunderbolt 3 audio interface also runs very well and stable.

**Comment by hksexybaby** [comment]
can share EFI for me .

**Reply by hksexybaby** [reply]
my nuc8i7hvk hackintosh upgrade to sonoma is black screen.

**Reply by SashonIT** [reply]
https://github.com/osy/HaC-Mini/discussions/1095#discussioncomment-7788717


Its for 14.1.2 without a Serial. You need to generate one.

I dont know your Background! 

Are you familiar with all these Hackintosh Tools?

You need to put a little work on this EFI.  Its not only copying and than ist done.

It needs Effort and ambitions from yourself.

Also i recommend to read and read and read the following Document. ---> https://dortania.github.io/OpenCore-Install-Guide/troubleshooting/extended/post-issues.html#disabling-sip

**Reply by hksexybaby** [reply]
Hi SashonIT
my nuc8i7hvk hackintosh from 13.1 upgrade 14.1
but black screen.

**Reply by SashonIT** [reply]
There is a massive lack of Information from your side.
Shall i guess now whats wrong with your maschine?
Under These circumstances i cannot help.

!My Car is green but Red traffic Light!

You cannot update to sonoma out of the Ventura OS. You need to make a Clean Sonoma Install. There is the EFI for.

Its a complete new System ---- > But you can roll backwards...... which means this EFI runs also good with Ventura.

After a sucesful installation you need to Patch the System to have WIFI.

It seems to be that you are new to this field of Hackintoshing.

So do me a favour and Read The F..... Manuals about it.

**Comment by 1git** [comment]
I noticed an issue with 14.4 update... The boot screen only gets half way then freezes... anyone else noticed this? 
Has anyone found a fix?

**Reply by SashonIT** [reply]
Read this here first.  Use deepl Software to translate into your Language.  https://www.hackintosh-forum.de/forum/thread/58656-anleitung-f%C3%BCr-broadcom-wlan-patch-unter-macos-sonoma/?postID=788294#post788294

You need  this here ….——-> With the corresponding IOSkywalkFamily.kext (vers. 1.1.0) exchange and the new OCLP 1.4.1 root patch, everything worked fine after the update to MacOS 14.4. WLAN is still/again there and works as it should. Thanks to the forum.

Also the OCLP is now updated to Version 1.4.1.  

Read this here. https://github.com/dortania/OpenCore-Legacy-Patcher/releases


@1git  Please do us a favour and report also your success and how you did it, as only reporting Problems.  We also want to read the good and healthy News here.

**Comment by Malumen** [comment]
Can I use this `.pkg` to update to `macOS Ventura`?

**Comment by happycodexing** [comment]
Based on the latest release,The default brightness is the lowest when the computer is turned on, no matter how much it is adjusted after the computer is turned on. thank you very much for your contribution! the another question is my LG ultrafine Display can not adjust volume,The sound can be made, but the slider to adjust the volume is gray.i would appreciate for your help.

**Comment by errorstack** [comment]
Thank you @goodbest , works really well for me!

**Comment by 1git** [comment]
All seems to still be working as before:
both 14.4.1 - 14.6.1
And Sequoia 15.0 GA
no Phone Mirroring though, but that is a T2/Apple Silicon only feature...
![image](https://github.com/user-attachments/assets/0ca44f6a-83dc-47c1-8152-b41bac230f56)

**Comment by C6JW** [comment]
Sharing my EFI for people with Intel wifi cards
https://github.com/C6JW/Hacmini/tree/main
You have to add your Serial codes
